provider-service/vai/internal/vai_event_flow.go

```diff
@@ -0,0 +1,257 @@
+package internal
+
+import (
+	aiplatform "cloud.google.com/go/aiplatform/apiv1"
+	"cloud.google.com/go/aiplatform/apiv1/aiplatformpb"
+	"context"
+	"github.com/go-logr/logr"
+	"github.com/googleapis/gax-go/v2"
+	"github.com/sky-uk/kfp-operator/argo/common"
+	. "github.com/sky-uk/kfp-operator/provider-service/base/pkg"
+	"github.com/sky-uk/kfp-operator/provider-service/base/pkg/streams"
+)
+
+const (
+	PushedModelArtifactType        = "tfx.PushedModel"
+	ModelPushedMetadataProperty    = "pushed"
+	ModelPushedMetadataValue       = 1
+	ModelPushedDestinationProperty = "pushed_destination"
+)
+
+var labels = struct {
+	PipelineName              string
+	PipelineNamespace         string
+	PipelineVersion           string
+	RunConfigurationName      string
+	RunConfigurationNamespace string
+	RunName                   string
+	RunNamespace              string
+}{
+	PipelineName:              "pipeline-name",
+	PipelineNamespace:         "pipeline-namespace",
+	PipelineVersion:           "pipeline-version",
+	RunConfigurationName:      "runconfiguration-name",
+	RunConfigurationNamespace: "runconfiguration-namespace",
+	RunName:                   "run-name",
+	RunNamespace:              "run-namespace",
+}
+
+type VaiEventFlow struct {
+	ProviderConfig    VAIProviderConfig
+	PipelineJobClient PipelineJobClient
+	Logger            logr.Logger
```

Comment:
```suggestion
```
We should remove logger, its available within the context which is already provided within the struct